contracts/ethregistrar/ETHRegistrarController.sol

```diff
@@ -3,176 +3,312 @@ pragma solidity >=0.8.4;
     }
 
     function setPriceOracle(PriceOracle _prices) public onlyOwner {
         prices = _prices;
         emit NewPriceOracle(address(prices));
     }
 
-    function setCommitmentAges(uint _minCommitmentAge, uint _maxCommitmentAge) public onlyOwner {
-        minCommitmentAge = _minCommitmentAge;
-        maxCommitmentAge = _maxCommitmentAge;
+    function withdraw() public {
+        payable(owner()).transfer(address(this).balance);
     }
 
-    function withdraw() public onlyOwner {
-        payable(msg.sender).transfer(address(this).balance);        
+    function supportsInterface(bytes4 interfaceID)
+        external
+        pure
+        returns (bool)
+    {
+        return
+            interfaceID == INTERFACE_META_ID ||
+            interfaceID == COMMITMENT_CONTROLLER_ID ||
+            interfaceID == COMMITMENT_WITH_CONFIG_CONTROLLER_ID;
     }
 
-    function supportsInterface(bytes4 interfaceID) external pure returns (bool) {
-        return interfaceID == INTERFACE_META_ID ||
-               interfaceID == COMMITMENT_CONTROLLER_ID ||
-               interfaceID == COMMITMENT_WITH_CONFIG_CONTROLLER_ID;
+    function onERC1155Received(
+        address,
+        address,
+        uint256,
+        uint256,
+        bytes calldata
+    ) external pure returns (bytes4) {
+        //TODO: guard against any other contract sending erc1155 to it apart from NameWrapper
+        return
+            bytes4(
+                keccak256(
+                    "onERC1155Received(address,address,uint256,uint256,bytes)"
+                )
+            );
     }
 
-    function _consumeCommitment(string memory name, uint duration, bytes32 commitment) internal returns (uint256) {
+    /* Internal functions */
+
+    function _consumeCommitment(
+        string memory name,
+        uint256 duration,
+        bytes32 commitment
+    ) internal returns (Cost memory) {
         // Require a valid commitment
         require(commitments[commitment] + minCommitmentAge <= block.timestamp);
 
         // If the commitment is too old, or the name is registered, stop
         require(commitments[commitment] + maxCommitmentAge > block.timestamp);
         require(available(name));
 
-        delete(commitments[commitment]);
+        delete (commitments[commitment]);
+
+        Cost memory cost = rentPrice(name, duration);
 
-        uint cost = rentPrice(name, duration);
         require(duration >= MIN_REGISTRATION_DURATION);
-        require(msg.value >= cost);
 
         return cost;
     }
+
+    function _setRecords(
+        address resolver,
+        bytes32 label,
+        bytes[] calldata data
+    ) internal {
+        // use hardcoded .eth namehash
+        bytes32 nodehash = keccak256(abi.encodePacked(base.baseNode(), label));
+        for (uint256 i = 0; i < data.length; i++) {
+            // check first few bytes are namehash
+            bytes32 txNamehash = data[i].readBytes32(4);
```

Comment:
In solidity 0.8, the `readX` methods are obsolete, because we can do this:

```suggestion
            bytes32 txNamehash = bytes32(data[i][4:36])
```

Reply:
Didn't seem to work for me. I got:

```
TypeError: Explicit type conversion not allowed from "bytes calldata slice" to "bytes32".
bytes32 txNamehash = bytes32(data[i][4:36]);
```

Reply:
PTAL!